### PR TITLE
feat: add pre-commit hooks

### PR DESCRIPTION
**Pull request by Birdi7**
# Description

Add pre-commit hooks with black, isort, flake8 and some other useful tools.

### CODE REVIEW

**Comment by uwinx** [comment]
@Birdi7, how's the progress so far?

**Comment by Birdi7** [comment]
> @Birdi7, how's the progress so far?

The PR is fully done, as far as I can remember

**Comment by Birdi7** [comment]
> Can we add mypy hook and resolve conflicts?

Yes, of course! The conflicts are resolved, and the mypy hook was added in https://github.com/aiogram/aiogram/pull/281/commits/46957d731f69075ec280f67acaa53a2c09563274

**Comment by Birdi7** [comment]
I have no idea why CI failed. Do you have any suggestions?

**Comment by Birdi7** [comment]
> I have no idea why CI failed. Do you have any suggestions?

Okay, now everything is fine

File: .isort.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[settings]
+known_third_party = aiofiles,aiohttp,aiohttp_socks,aresponses,async_lru,pydantic,pytest,typing_extensions
```
Comment (JrooTJunior) [inline .isort.cfg:2]: Does it works with config from `pyproject.toml` at the same time?

**Comment by JrooTJunior** [comment]
Its really works with python 3.8? Or maybe i do something wrong?

`pre-commit install`
and then i'll try to execute manually `pre-commit` and got an error:
```
An unexpected error has occurred: CalledProcessError: command: ('/home/jrootjunior/.cache/pypoetry/virtualenvs/magic-filters-4RRMQLxm-py3.8/bin/python', '-mvirtualenv', '/home/jrootjunior/.cache/pre-commit/repo56k2ttud/py_env-python3.7', '-p', 'python3.7')
```

**Comment by JrooTJunior** [comment]
I test this config on another related project [aiogram/magic-filter](https://github.com/aiogram/magic-filter) and it works correctly. Thanks for an improvement!

**Pull request merged**